mlir-tensorrt/tensorrt/lib/Target/TranslateToTensorRT.cpp

```diff
@@ -434,6 +440,14 @@ tensorrt::buildFunction(mlir::FunctionOpInterface op,
              << "failed to set timing cache";
   }
 
+  // Attach a custom GPU allocator
+  constexpr uint64_t kCURR_ALLOC_SIZE{1073741824U};
+  std::unique_ptr<nvinfer1::IGpuAllocator> gpuAllocator(
+      new TensorRTLinearAllocator(kCURR_ALLOC_SIZE));
+  if (opts.useCustomGpuAllocator) {
+    builder->setGpuAllocator(gpuAllocator.get());
```

Comment:
@christopherbate Here are the builder-side changes to allow the GPU allocator to be set up. We probably do not need this and will only be required for runtime allocations.